### PR TITLE
fix: loading filters inplp

### PR DESCRIPTION
**Pull request by ArthurTriis1**
Signed-off-by: Arthur Andrade <arthurfelandrade@gmail.com>

## What's the purpose of this pull request?
<em>Considering the context, what is the problem we'll solve? Where in VTEX's big picture our issue fits in? Write a tweet about the context and the problem itself.</em>

## How it works? 
<em>Tell us the role of the new feature, or component, in its context.</em>

## How to test it?
<em>Describe the steps with bullet points. Is there any external reference, link, or example?</em> 

## References
<em>Spread the knowledge: is this any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: add references to related issues or mention people important to this PR may be good for the documentation and reviewing process</em>

### CODE REVIEW

**Review by filipewl — APPROVED** [review]
Tested at https://preview-261--base.preview.vtex.app/office/ and it's a very nice improvement that the filters are ordered and that they stay there as the filtering is happening.

CleanShot 2022-01-21 at 11 45 35

**Comment by filipewl** [comment]
Took a quick look at [the failing test](https://faststore-test-assets.s3.amazonaws.com/187364a780c1254419dcc0ffb10f9004#:~:text=Search%20page%20Filters%20and%20Sorting%20options) and it's simple to fix. Because the accordion stays open on the second opening, we need to expand it anymore.
CleanShot 2022-01-21 at 11 58 32

File: src/components/sections/ProductGallery/ProductGallery.tsx [hunk 1]
```diff
@@ -38,7 +40,25 @@ function ProductGallery({ title }: Props) {
...
+        }
+
+        return facet
+      })
+
+      facets.current = orderFacets
+    }
+
+    return facets.current
+  }, [data])
```
Comment (danzanzini) [inline src/components/sections/ProductGallery/ProductGallery.tsx:59]: We should avoid this kind of processing in the front-end. We don't need to tackle this now but I think it's good to create a Jira ticket to try to send this behavior to the server.

**Review by danzanzini — APPROVED** [review]
Nice improvement! Approved

**Comment by filipewl** [comment]
[Squashed](bc6480990add7f517a45b9764871239697e2d213) directly to `feat/style-plp`'s #240 because a failing test from the target branch prevented the merge here.

**Review by eduardoformiga — COMMENTED** [review]
Woww, I parachuted here just now. Amazing trick! ✨

**Pull request closed without merging**